python/test/unit/geometry/test_bounding_box_tree.py

```diff
@@ -496,3 +497,82 @@ def test_surface_bbtree_collision(dtype):
 
     collisions = compute_collisions_trees(bbtree1, bbtree2)
     assert len(collisions) == 1
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_determine_point_ownership(dtype):
+    """Find point owners (ranks and cells) using bounding box trees + global communication
+    and compare to point ownership data results."""
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    tdim = 2
+    num_cells_side = 4
+    h = dtype(1.0 / num_cells_side)
+    mesh = create_unit_square(MPI.COMM_WORLD,
+                              num_cells_side, num_cells_side,
+                              CellType.quadrilateral, dtype=dtype)
+    cell_map = mesh.topology.index_map(tdim)
+
+    def test_po_against_bb_tree(mesh,points,cells):
+        cells = np.array([cell for cell in cells if cell < cell_map.size_local])
+        num_points = points.shape[0]
+        po = determine_point_ownership(mesh,points,cells)
+        tree = bb_tree(mesh, mesh.topology.dim, cells)
+        assert(mesh.topology.cell_type in [CellType.quadrilateral,CellType.hexahedron])
+        # Since box elements are considered, bounding box collisions correspond
+        # to the actual collisions
+        tree_col = compute_collisions_points(tree, points)
+        local_owners = np.zeros(2*num_points,dtype=np.int32) # rankwise data before Allreduce
+        owners = local_owners.copy()
+        for ipoint in range(num_points):
+            cell = tree_col.links(ipoint)
+            if len(cell) > 0:
+                local_owners[2*ipoint] = rank
+                local_owners[2*ipoint+1] = cell[0]
+        comm.Allreduce(local_owners, owners, op=MPI.SUM)
+        owner_ranks = owners[[2*i   for i in range(num_points)]]
+        owner_cells = owners[[2*i+1 for i in range(num_points)]]
+
+        assert((po.src_owner() == owner_ranks).all())
+
+        dest_points = po.dest_points()
+        # Since all processors ask about the same points,
+        # dest_points contains repeated data:
+        # dest_points = [p0, p1, p2, p3, ..., pN] * comm.size()
+        ipoint = 1
+        duplicate_start_index = -1
+        while (ipoint < dest_points.shape[0]) and not(duplicate_start_index >= 0):
+            if ((dest_points[ipoint,:] == dest_points[0,:]).all()):
+                duplicate_start_index = ipoint
+            ipoint += 1
+        dest_points = dest_points[range(duplicate_start_index)]
+        dest_cells  = po.dest_cells()[range(duplicate_start_index)]
+
+        # For the points owned by the current rank, compare owner cells
+        search_indices_dest_points = list(range(len(points)))
+        for ipoint in range(dest_points.shape[0]):
+            dest_point = dest_points[ipoint,:]
+            # Find corresponding index in `points`
+            jpoint = -1
+            for jpoint in search_indices_dest_points:
+                point = points[jpoint,:]
+                if (np.isclose(np.linalg.norm(dest_point-point),0.0)):
```

Comment:
Here we probably need a atol dependning on the floating point tolerance used for the mesh geometry, see for instance: https://github.com/FEniCS/dolfinx/blob/06e90b8506d967f1836ffac7f289f49cb0fd17ec/python/test/unit/fem/test_symmetry.py#L42